tests/functional/UseCase/Bucket/ShareLinkBucketTest.php

```diff
@@ -0,0 +1,359 @@
+
+        $handler(
+            $this->getCredentials(),
+            $command,
+            []
+        );
+
+        $sourceBqClient = $this->clientManager->getBigQueryClient($this->sourceProjectCredentials);
+
+        $handler = new CreateTableHandler($this->clientManager);
+        $path = new RepeatedField(GPBType::STRING);
+        $path[] = $bucketDatabaseName;
+        $columns = new RepeatedField(GPBType::MESSAGE, CreateTableCommand\TableColumn::class);
+        $columns[] = (new CreateTableCommand\TableColumn())
+            ->setName('ID')
+            ->setType(Bigquery::TYPE_INTEGER);
+        $command = (new CreateTableCommand())
+            ->setPath($path)
+            ->setTableName('TESTTABLE_AFTER')
+            ->setColumns($columns);
+        $handler(
+            $this->sourceProjectCredentials,
+            $command,
+            []
+        );
+        // check that there is no need to re-share or whatever
+        $sourceBqClient->runQuery($sourceBqClient->query(sprintf(
+            'INSERT INTO %s.`TESTTABLE_AFTER` (`ID`) VALUES (1)',
+            BigqueryQuote::quoteSingleIdentifier($bucketDatabaseName)
+        )));
+
+        $analyticHubClient = $this->clientManager->getAnalyticHubClient($this->getCredentials());
+
+        $formattedName = $analyticHubClient->listingName(
+            $sourceProjectId,
+            GCPClientManager::DEFAULT_LOCATION,
+            $this->sourceProjectResponse->getProjectReadOnlyRoleName(),
+            $bucketDatabaseName
+        );
+        $listing = $analyticHubClient->getListing($formattedName);
+        $this->assertNotNull($listing->getName());
+
+        $publicPart = (array) json_decode(
+            $this->targetProjectResponse->getProjectUserName(),
+            true,
+            512,
+            JSON_THROW_ON_ERROR
+        );
+        /** @var string $targetProjectId */
+        $targetProjectId = $publicPart['project_id'];
+        $linkedBucketSchemaName = $bucketDatabaseName . '_LINKED';
+
+        $handler = new LinkBucketHandler($this->clientManager);
+        $command = (new LinkBucketCommand())
+            ->setStackPrefix($this->getStackPrefix())
+            ->setTargetProjectId($targetProjectId)
+            ->setTargetBucketId($linkedBucketSchemaName)
+            ->setSourceShareRoleName($listing->getName()); // listing
+
+        $handler(
+            $this->getCredentials(),
+            $command,
+            []
+        );
+
+        $handler = new UnShareBucketHandler($this->clientManager);
+        $command = (new UnShareBucketCommand())
+            ->setBucketShareRoleName($listing->getName());
+
+        $handler(
+            $this->getCredentials(),
+            $command,
+            []
+        );
+
+        //@todo drop lisnting do not drop the linked dataset or fail, we can list all
+        // linked datasets and drop it but is not an atomic operation
+    }
+
+
+    public function tmpCreateShareRoleNameForBucket(
```

Comment:
to nevypada, ze by se to nekde pouzivalo a pokud jo, tak se ptam proc a nemi tam normalni generator -> muzeme to zrusit?

Reply:
Dik, neviem preco mi to nezosedlo ze sa to nikde nepouziva zmazal som to